### PR TITLE
Add unit tests for processing job

### PR DESCRIPTION
**Pull request by ryansteakley**
Description of changes:

- `Sdk.go` coverage 72.0%
- Custom code hook coverage 100%

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                }
+            },
+            {
+                "AppManaged": false,
+                "FeatureStoreOutput": null,
+                "OutputName": "test_data",
+                "S3Output": {
+                    "LocalPath": "/opt/ml/processing/output_test/",
+                    "S3UploadMode": "EndOfJob",
+                    "S3Uri": "s3://source-data-bucket-592697580195-us-west-2/sagemaker/processing/output/"
```
Comment (mbaijal) [inline pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json:74]: looks like an account id that we should edit out

File: pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        "KmsKeyId": null,
+        "Outputs": [
+            {
+                "AppManaged": false,
+                "FeatureStoreOutput": null,
+                "OutputName": "train_data",
+                "S3Output": {
+                    "LocalPath": "/opt/ml/processing/output_train/",
+                    "S3UploadMode": "EndOfJob",
+                    "S3Uri": "s3://source-data-bucket-592697580195-us-west-2/sagemaker/processing/output/"
```
Comment (mbaijal) [inline pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json:64]: Remove account id

File: pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                }
+            },
+            {
+                "AppManaged": false,
+                "FeatureStoreOutput": null,
+                "OutputName": "valid_data",
+                "S3Output": {
+                    "LocalPath": "/opt/ml/processing/output_valid/",
+                    "S3UploadMode": "EndOfJob",
+                    "S3Uri": "s3://source-data-bucket-592697580195-us-west-2/sagemaker/processing/output/"
```
Comment (mbaijal) [inline pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json:84]: same comment

File: pkg/resource/processing_job/testdata/sdkapi/describe/success_completed_describe.json [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            "VolumeKmsKeyId": null,
+            "VolumeSizeInGB": 20
+        }
+    },
+    "ProcessingStartTime": null,
+    "RoleArn": "arn:aws:iam::123456789012:role/service-role/AmazonSageMaker-ExecutionRole-20210920T111639",
+    "StoppingCondition": {
+        "MaxRuntimeInSeconds": 86400
+    },
+    "TrainingJobArn": null
```
Comment (mbaijal) [inline pkg/resource/processing_job/testdata/sdkapi/describe/success_completed_describe.json:102]: Does this mean we arent testing that a successfully completed job has an ARN ? - can we add this as a comment in the test for future reference

File: pkg/resource/processing_job/testdata/test_suite.yaml [hunk 5]
```diff
@@ -1,17 +1,174 @@
...
+              error:
+                code: ValidationException
+                message: "Could not find requested job"
+        invoke: Delete
+        expect:
+          error: nil      
+      - name: "Delete=Successful"
+        description: "This test checks if the Processing job returns nil to mark the resource Unmanaged if job not in InProgress state"
+        given:
+          desired_state: "v1alpha1/readone/observed/completed.yaml"
```
Comment (mbaijal) [inline pkg/resource/processing_job/testdata/test_suite.yaml:169]: Shouldn't this be tested on a job which is in-progress ?

File: pkg/resource/processing_job/testdata/test_suite.yaml [hunk 6]
```diff
@@ -1,17 +1,174 @@
...
+        given:
+          desired_state: "v1alpha1/create/desired/success_before_create.yaml"
+          svc_api:
+            - operation: CreateProcessingJobWithContext
+              output_fixture: "sdkapi/create/success_create.json"
+        invoke: Create
+        expect:
+          latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+          error: nil
+      - name: "Create=Valid2"
```
Comment (mbaijal) [inline pkg/resource/processing_job/testdata/test_suite.yaml:29]: Can we name such that the variation is apparent ?

File: pkg/resource/processing_job/testdata/test_suite.yaml [hunk 7]
```diff
@@ -1,17 +1,174 @@
-tests:
+  tests:
   - name: "Processing job create tests"
     description: "Part of processing job CRD tests."
```
Comment (mbaijal) [inline pkg/resource/processing_job/testdata/test_suite.yaml:3]: Edit this description to be more meaningful and in sync with the other tests here

File: pkg/resource/processing_job/testdata/test_suite.yaml [hunk 8]
```diff
@@ -1,17 +1,174 @@
...
+        description: "Testing readOne when required field is missing. No API call is made and returns error."
+        given: 
+          desired_state: "v1alpha1/readone/desired/missing_required_field.yaml"
+        invoke: ReadOne
+        expect:
+          error: "resource not found"
+      - name: "ReadOne=NotFound"
+        description: "Testing readOne when Describe fails to find the resource on SageMaker"
+        given: 
+          desired_state: "v1alpha1/readone/desired/after_create.yaml"
```
Comment (mbaijal) [inline pkg/resource/processing_job/testdata/test_suite.yaml:53]: nit: after_create.yaml is not a clear enough filename IMO, not sure what it means

**Review by mbaijal — CHANGES_REQUESTED** [review]

File: pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json (hunk 1 above)
Comment (ryansteakley) [reply pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json:74]: its the marin-public-test-data account, used in other unit tests as well. I think it should be fine?

File: pkg/resource/processing_job/testdata/sdkapi/describe/success_completed_describe.json (hunk 4 above)
Comment (ryansteakley) [reply pkg/resource/processing_job/testdata/sdkapi/describe/success_completed_describe.json:102]: We are checking for the processing_job_arn, in this example the processingJob doesn't run trainingjobs so I believe trainingjobArn should be null.

File: pkg/resource/processing_job/testdata/test_suite.yaml (hunk 5 above)
Comment (ryansteakley) [reply pkg/resource/processing_job/testdata/test_suite.yaml:169]: changed the test names

File: pkg/resource/processing_job/testdata/test_suite.yaml (hunk 8 above)
Comment (ryansteakley) [reply pkg/resource/processing_job/testdata/test_suite.yaml]: changed file name

File: pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json (hunk 1 above)
Comment (mbaijal) [reply pkg/resource/processing_job/testdata/sdkapi/describe/stopping_describe.json:74]: ok, thanks for confirming.

File: pkg/resource/processing_job/testdata/sdkapi/describe/success_completed_describe.json (hunk 4 above)
Comment (mbaijal) [reply pkg/resource/processing_job/testdata/sdkapi/describe/success_completed_describe.json:102]: got it.

File: pkg/resource/processing_job/testdata/test_suite.yaml (hunk 5 above)
Comment (mbaijal) [reply pkg/resource/processing_job/testdata/test_suite.yaml:169]: thanks.

**Review by mbaijal — APPROVED** [review]

**Pull request merged**